### ISSUE TITLE
New formula for raid parties created / joined in leaderboards

### ISSUE THREAD

**Issue by maconleary** [PokeNavBot/Issue-Tracker#541]
**Is your feature request related to a problem? Please describe.**
To be accurate, the existing formulae assume that everyone would leave unsuccessful raids prior to expiry or $end.
This assumption doesn't hold for my server, so I don't use either of these leaderboards as they would overcount raids.

**Describe the solution you'd like**
Now that the $go command exists, I was thinking that usage of this would be a much more reliable indicator that a raid happened.
I'd like to add an extra condition to the parties created / joined rules: ($go command was used), if this is technically possible.

**Comment by TheGerontocracy** [comment]
I like the idea, but I would count at $go instead to eliminate the costs of exiting the raid-channel before expiry (a bigger issue to me). 

For raids like Timburr, there are often multiple $go, so counting at $go would need to account for that.

Something along these lines: 

```
# At channel creation
eligible_members = {}
# At each $go in the raid-channel: 
eligible_members.update(raid_role.members)
# At expiry:
if len(eligible_members) > 1:
    ... # add 1 to raids-joined of each member.
    ... # add 1 to raids-created for the creator. 
```

**Comment by maconleary** [comment]
Ha. I had completely removed a eligible_member component originally and then I got worried that it was too subject to manipulation if you could trigger it with just a single account and a fake $go (legendaries in particular). 

Does your version mean that you'd need to have 2+ single person raid parties to trigger it? It's not perfect, but it's better than what I had, I think.

**Comment by TheGerontocracy** [comment]
Yeah, that’s also the current condition. That’s why I’m not too worried about overcounting right now. If at least two members sign on, I’m fine with granting points for effort.

**Comment by Jahrern** [comment]
A mod channel where the bot asks if the raid was done could work. An opt-in alternative for those that want their leaderboards to be more accurate. A simple up/down reaction could be the yes/no alternative. Honestly think that would work.

**Comment by maconleary** [comment]
I think it might work on _some_ servers, but reckon it'd be too much mod input required for many.

**Comment by Jahrern** [comment]
> I think it might work on _some_ servers, but reckon it'd be too much mod input required for many.

In all honesty, the effort is pretty minimal after the fact. It would require you to pay attention to raid channels if they were actually completed or not. That is definetly work, but a good mod team already have most of that covered. This is just an easy to implement option for those that want more control. I'm all for an easier implementation though. But this is the only bulletproof option i could come up with without being a coder.

**Comment by FlipsCat** [comment]
In my opinion, it all cancels out in the end. I've hosted probably hundreds of T1s by now where only 1 joined. alternatively, i've attempted to host probably hundreds of T5s that got 3 or 4 members and never had enough to actually lobby. In the end users realize that the numbers are basically accurate for those actually hosting raids at the top.
